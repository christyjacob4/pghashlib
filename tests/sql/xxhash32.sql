-- Test basic hash functionality with text
SELECT xxhash32('hello world');

-- Test hash with different text inputs
SELECT xxhash32('test string');
SELECT xxhash32('another test');

-- Test text input with custom seed
SELECT xxhash32('hello world', 42);
SELECT xxhash32('hello world', 123);

-- Test bytea input
SELECT xxhash32('hello world'::bytea);

-- Test bytea input with custom seed
SELECT xxhash32('hello world'::bytea, 42);

-- Test integer input
SELECT xxhash32(12345);
SELECT xxhash32(-12345);

-- Test integer input with custom seed
SELECT xxhash32(12345, 42);
SELECT xxhash32(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT xxhash32('consistent test') = xxhash32('consistent test');

-- Test seed effect (same input, different seeds should give different hashes)
SELECT xxhash32('seed test', 1) != xxhash32('seed test', 2);

-- Test empty string
SELECT xxhash32('');

-- Test single character
SELECT xxhash32('a');

-- Test long string
SELECT xxhash32('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'xxhash32'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';