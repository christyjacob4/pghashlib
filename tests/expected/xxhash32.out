-- Test basic hash functionality with text
SELECT xxhash32('hello world');
  xxhash32  
------------
 -826579422
(1 row)

-- Test hash with different text inputs
SELECT xxhash32('test string');
 xxhash32  
-----------
 633576710
(1 row)

SELECT xxhash32('another test');
 xxhash32  
-----------
 107070056
(1 row)

-- Test text input with custom seed
SELECT xxhash32('hello world', 42);
 xxhash32  
-----------
 -69933708
(1 row)

SELECT xxhash32('hello world', 123);
  xxhash32   
-------------
 -1375582818
(1 row)

-- Test bytea input
SELECT xxhash32('hello world'::bytea);
  xxhash32  
------------
 -826579422
(1 row)

-- Test bytea input with custom seed
SELECT xxhash32('hello world'::bytea, 42);
 xxhash32  
-----------
 -69933708
(1 row)

-- Test integer input
SELECT xxhash32(12345);
  xxhash32  
------------
 -393624961
(1 row)

SELECT xxhash32(-12345);
  xxhash32  
------------
 1221658118
(1 row)

-- Test integer input with custom seed
SELECT xxhash32(12345, 42);
  xxhash32   
-------------
 -1202603880
(1 row)

SELECT xxhash32(-12345, 123);
  xxhash32  
------------
 1538418724
(1 row)

-- Test consistency (same input should give same hash)
SELECT xxhash32('consistent test') = xxhash32('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test seed effect (same input, different seeds should give different hashes)
SELECT xxhash32('seed test', 1) != xxhash32('seed test', 2);
 ?column? 
----------
 t
(1 row)

-- Test empty string
SELECT xxhash32('');
 xxhash32 
----------
 46947589
(1 row)

-- Test single character
SELECT xxhash32('a');
  xxhash32  
------------
 1426945110
(1 row)

-- Test long string
SELECT xxhash32('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.');
 xxhash32  
-----------
 375463304
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'xxhash32'
ORDER BY proname, proargtypes;
 proname  | provolatile | proisstrict 
----------+-------------+-------------
 xxhash32 | i           | t
 xxhash32 | i           | t
 xxhash32 | i           | t
 xxhash32 | i           | t
 xxhash32 | i           | t
 xxhash32 | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)
